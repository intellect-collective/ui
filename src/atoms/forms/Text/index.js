import React from 'react';
import changeable from '../_decorators/changeable';
import formContext from '../_decorators/formContext';
import { compose } from '../_decorators/_utils';

const Text = (props) => (
    <input { ...props } type="text" />
);
Text.displayName = 'Text';

export default compose(
    formContext(),
    changeable()
)(Text);
