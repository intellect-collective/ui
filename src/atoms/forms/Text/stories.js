import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import Text from './index';

storiesOf('Atoms / Forms / Text', module)
    .add('default', () => (<Text name="test-text-1" onChange={ action('onChange') } />));
