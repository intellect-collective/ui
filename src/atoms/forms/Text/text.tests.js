import React from 'react';
import Text from '.';
import {
    basicChangeableTests
} from '../../../../test/utils';

const validProps = {
    name: 'test',
    value: 'pithy',
    onChange: () => {}
};
const expectedProps = {
    name: 'test',
    value: 'pithy',
    onChange: expect.any(Function)
};

describe('Text', () => {
    basicChangeableTests(Text, 'input', validProps, expectedProps);
});
