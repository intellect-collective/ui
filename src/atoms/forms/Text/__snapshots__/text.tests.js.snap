// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text Should accept valid props 1`] = `
<FormContext
  name="test"
  onChange={[Function]}
  value="pithy"
>
  <Changeable
    name="test"
    onChange={[Function]}
    value="pithy"
  >
    <Text
      name="test"
      onChange={[Function]}
      value="pithy"
    >
      <input
        name="test"
        onChange={[Function]}
        type="text"
        value="pithy"
      />
    </Text>
  </Changeable>
</FormContext>
`;
